### ISSUE TITLE
TS Server fatal error:  Debug Failure.

### ISSUE THREAD

**Issue by maldan** [microsoft/vscode#117101]
Issue Type: <b>Bug</b>

❗️❗️❗️ Please fill in the sections below to help us diagnose the issue ❗️❗️❗️

**TypeScript Version:** 4.1.5

**Steps to reproduce crash**
1.
2.
3.

**TS Server Log**
❗️ Please review and upload this log file to help us diagnose this crash:
`c:\Users\black\AppData\Roaming\Code\logs\20210220T020144\exthost1\vscode.typescript-language-features\tsserver-log-Mui4EL\tsserver.log`
The log file may contain personal data, including full paths and source code from your workspace. You can scrub the log file to remove paths or other personal information.

**TS Server Error Stack**
Server: `syntax`
```
Error: Debug Failure.
    at actualResolveModuleNamesWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:105104:161)
    at resolveModuleNamesWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:105342:26)
    at resolveModuleNamesReusingOldState (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:105532:19)
    at processImportedModules (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106943:35)
    at findSourceFileWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106715:17)
    at findSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106577:26)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:85
    at getSourceFileFromReferenceWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106501:34)
    at processSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:13)
    at processRootFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106344:13)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106913:21
    at Object.forEach (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:188:30)
    at processLibReferenceDirectives (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106908:16)
    at findSourceFileWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106712:21)
    at findSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106577:26)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:85
    at getSourceFileFromReferenceWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106501:34)
    at processSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:13)
    at processRootFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106344:13)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106913:21
    at Object.forEach (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:188:30)
    at processLibReferenceDirectives (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106908:16)
    at findSourceFileWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106712:21)
    at findSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106577:26)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:85
    at getSourceFileFromReferenceWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106501:34)
    at processSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:13)
    at processRootFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106344:13)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106913:21
    at Object.forEach (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:188:30)
    at processLibReferenceDirectives (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106908:16)
    at findSourceFileWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106712:21)
    at findSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106577:26)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:85
    at getSourceFileFromReferenceWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106501:34)
    at processSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:13)
    at processRootFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106344:13)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106913:21
    at Object.forEach (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:188:30)
    at processLibReferenceDirectives (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106908:16)
    at findSourceFileWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106712:21)
    at findSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106577:26)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:85
    at getSourceFileFromReferenceWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106501:34)
    at processSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:13)
    at processRootFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106344:13)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106913:21
    at Object.forEach (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:188:30)
    at processLibReferenceDirectives (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106908:16)
    at findSourceFileWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106712:21)
    at findSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106577:26)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:85
    at getSourceFileFromReferenceWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106501:34)
    at processSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:13)
    at processRootFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106344:13)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106913:21
    at Object.forEach (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:188:30)
    at processLibReferenceDirectives (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106908:16)
    at findSourceFileWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106712:21)
    at findSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106577:26)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:85
    at getSourceFileFromReferenceWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106501:34)
    at processSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:13)
    at processRootFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106344:13)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106913:21
    at Object.forEach (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:188:30)
    at processLibReferenceDirectives (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106908:16)
    at findSourceFileWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106712:21)
    at findSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106577:26)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:85
    at getSourceFileFromReferenceWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106501:34)
    at processSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:13)
    at processRootFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106344:13)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106913:21
    at Object.forEach (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:188:30)
    at processLibReferenceDirectives (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106908:16)
    at findSourceFileWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106712:21)
    at findSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106577:26)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:85
    at getSourceFileFromReferenceWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106501:34)
    at processSourceFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106534:13)
    at processRootFile (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:106344:13)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:105209:60
    at Object.forEach (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:188:30)
    at Object.createProgram (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:105209:16)
    at synchronizeHostData (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:146572:26)
    at Proxy.getProgram (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:146664:13)
    at InferredProject.Project.updateGraphWorker (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:151530:53)
    at InferredProject.Project.updateGraph (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:151463:42)
    at updateProjectIfDirty (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:152932:45)
    at Array.forEach (<anonymous>)
    at ProjectService.assignProjectToOpenedScriptInfo (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:155109:41)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:155395:97
    at Object.flatMap (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:513:25)
    at ProjectService.applyChangesInOpenFiles (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:155395:41)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:156413:46)
    at c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:158286:88
    at IOSession.Session.executeWithRequestId (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:158277:28)
    at IOSession.Session.executeCommand (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:158286:33)
    at IOSession.Session.onMessage (c:\Users\black\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:158310:35)
```

VS Code version: Code 1.53.2 (622cb03f7e070a9670c94bae1a45d78d7181fbd4, 2021-02-11T11:48:04.245Z)
OS version: Windows_NT x64 10.0.19041

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3600)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.90GB (17.50GB free)|
|Process Argv|D:\\deno_lib\\remote-gl --crash-reporter-id b3f630f9-8e0c-40eb-9833-8e51f42bddb0|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (15)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-deno|den|2.3.3
vsc-material-theme|Equ|33.1.2
vsc-material-theme-icons|equ|1.2.2
prettier-vscode|esb|5.9.2
code-runner|for|0.11.2
vscode-env|Iro|0.1.0
vscode-docker|ms-|1.10.0
remote-containers|ms-|0.158.0
remote-wsl|ms-|0.53.4
hexeditor|ms-|1.4.0
vetur|oct|0.32.0
material-icon-theme|PKi|4.5.0
languagetool|ray|1.0.1
LiveServer|rit|5.6.1
es6-string-html|Tob|2.1.0

(2 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
vstry244cf:30256637
pythonvsdeb440:30248342
pythonvsded773:30248341
pythonvspyt875:30251590

```

</details>

**Comment by mjbvz** [comment]
Does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/) with all [extensions disabled](https://github.com/microsoft/vscode/wiki/Performance-Issues#run-with-extensions-disabled)? Specifically try disabling deno

**Comment by maldan** [comment]
I think I found the problem. It was deno after all. To be precise it was "deno.unstable" and "deno.lint" options in settings.json. When I set these options to "false" it stopped crashing. Before that ts-server randomly crashed each 10-20 minutes. But the problem is not it crashed so often. The problem is I can't restart it manually. When vscode showing me the message "Ts server crashed 5 times and won't restart..." or something like that. I trying to restart it by "Ctrl + Shift + P" -> "TypeScript: Restart TS Server" but nothing happens... I need to close vscode and open it again and then everything is works again. Is it a bug or feature? Why can't I restart it manually?
